From 658957a94f2c290bbc180f24aeafbca4ac576d1f Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 27 Feb 2014 23:21:00 +0000
Subject: [PATCH 4/5] Win32: Add some path checks to sh.exe as getSUCommand()
 for _WIN32

Relative path, working directory and then some hardcoded paths are
checked.
---
 src/wmhelper.cpp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/src/wmhelper.cpp b/src/wmhelper.cpp
index 187087e..6ea6fa2 100644
--- a/src/wmhelper.cpp
+++ b/src/wmhelper.cpp
@@ -292,6 +292,24 @@ QString WMHelper::getGKSUCommand(){
 QString WMHelper::getSUCommand(){
   QString result(ctn_NO_SU_COMMAND);
 
+#if defined(_WIN32)
+  QString testPath = QCoreApplication::applicationDirPath() + QLatin1String("/../../bin/sh.exe");
+  if (QFile::exists(testPath))
+  {
+    return testPath + QLatin1String(" -c ");
+  }
+  /* For use when hacking on this, could remove finally. */
+  testPath = QDir::currentPath() + QLatin1String("/sh.exe");
+  if (QFile::exists(testPath))
+  {
+    return testPath + QLatin1String(" -c ");
+  }
+  // Return a semi reasonable default.
+#if Q_OS_WIN32
+  return QLatin1String("/msys32/usr/bin/sh.exe -c ");
+#else
+  return QLatin1String("/msys64/usr/bin/sh.exe -c ");
+#endif
   if (isXFCERunning() && (UnixCommand::hasTheExecutable(ctn_GKSU_2))){
     result = getGKSUCommand();
   }
-- 
2.1.0

